TypeScript_ESLint.snb.md

```diff
@@ -0,0 +1,189 @@
+# How TypeScript ESLint Works, a source-level introduction
+
+![TypeScript and ESLint logos](https://user-images.githubusercontent.com/1646931/169179393-3f1db84a-080a-4887-9f9f-78ab9dc29d4c.png)
+
+[TypeScript ESLint](https://typescript-eslint.io) is the de facto tool for linting in TypeScript.
+It's used by many thousands of projects and ensures code quality, security, and best practices across the TypeScript world.
+If you're deploying TypeScript into production, chances are you're using this in your toolchain.
+
+In this post, we provide a hands-on, technical introduction to how this essential static analysis tool works by tracing through its key code paths.
+We'll cover the following:
+
+- How ESLint rules are defined
+- How ESLint rules use ASTs to analyze code files
+- Switching between ESLint's and TypeScript's AST formats
+- Calling TypeScript's type checking APIs to analyze code
+
+> Not familiar with ASTs yet?
+> No worries!
+> Read [TypeScript ESLint's Abstract Syntax Trees (ASTs) docs page](https://typescript-eslint.io/docs/development/architecture/asts) to learn how programs represent and reason about source code.
+
+Read on!
+
+## Overview
+
+TypeScript ESLint is implemented as a collection of packages that build on [ESLint](https://eslint.org).
+The packages we'll be going over in this walkthrough are:
+
+- [`eslint-plugin`](https://www.npmjs.com/package/@typescript-eslint/eslint-plugin), which defines the ESLint plugin and the linting rules.
+- [`parser`](https://www.npmjs.com/package/@typescript-eslint/parser) and [`typescript-estree`](https://www.npmjs.com/package/@typescript-eslint/typescript-estree), which are responsible for parsing TypeScript source code into an ESLint-compatible form, mapping from TypeScript ASTs to ESLint's AST representation (ESTree) and providing an API to query the AST from the rules.
+
+### ESLint Rules
+
+The purpose of a linter is to detect a set of patterns in source code that match common anti-patterns.
+ESLint, like many linters, defines each anti-pattern to search for in **rules**.
```

Comment:
Nit: grammar oddity. Since each anti-pattern is its own rule, it's slightly off to say "each" anti-pattern has "rules" (plural).

```suggestion
ESLint, like many linters, defines anti-patterns to search for in **rules**.
Each individual rule defines the logic for detecting and reporting on one common class of anti-pattern.
```